File: be/Forum/Forum.Data/Entities/Post.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Forum.Data.Entities
{
    public class Post
    {
        [Key] public int PostId { get; set; }

        public string Content { get; set; }

        public DateTime? CreatedAt { get; set; }

        public DateTime? EditedAt { get; set; }

        public string UserId { get; set; }

        public User User { get; set; }

        public int ThreadId { get; set; }

        public Thread Thread { get; set; }
    }
}

File: fe/client/src/store/reducers/post.ts
import { POST_ACTIONS } from "../actions/post"
import { USER } from "./auth"

export type POST = {
    postId: number,
    content: string,
    createdAt: Date,
    editedAt: Date,
    userId: string,
    user: USER,
    threadId: number
}

const INIT_STATE: {
    data: POST[],
} = {
    data: [],
}

export const postReducer = (state = INIT_STATE, action: { type: POST_ACTIONS, payload: { posts: POST[] } }) => {
    switch (action.type) {
        case POST_ACTIONS.GET:
            return { ...state, data: action.payload.posts }
        default:
            return state
    }
}